### DISCUSSION TITLE
LDAP calls not showing on transactions

### DISCUSSION THREAD

**Discussion by davrichards** [open-telemetry/opentelemetry-java-instrumentation#7607] [Q&A]
Hi in our location we have looked at using the otel java agent v1.22 on a 3rd party application. We are using elastic as the observability platform. In our testing we have seen that calls during login to ldap (MS active directory) are not being detected and not seen in transactions, dependency or service map.
Is this a known issue and if so is this to be resolved in future releases ?

**Comment by mateuszrzeszutek** [comment]
Hey @davrichards ,

The OTel javaagent currently does not have any LDAP API/client instrumentation.

My best suggestion is to use [manual instrumentation](https://opentelemetry.io/docs/instrumentation/java/manual/#acquiring-a-tracer) around the LDAP calls in your application.

**Reply by davrichards** [reply]
Hi this is a 3rd party application and we don't have any access to the source code so we can't modify it. We are relying on the agent implementation to expose all the calls to backend dependencies. Thanks.

**Reply by mateuszrzeszutek** [reply]
I see. In that case, you can use [this config option](https://opentelemetry.io/docs/instrumentation/java/automatic/annotations/#creating-spans-around-methods-with-otelinstrumentationmethodsinclude) to specify which methods should be instrumented with a "basic" span (doesn't have almost any attributes) -- you have to know the methods you want to decorate though.

Also, if you know which LDAP library is used in your application (there's several of them), you can open an enhancement issue -- while the maintainers of this project most likely won't find the time to implement a new instrumentation (there're very few of us and we have very limited time and resources), perhaps somebody from the community will find that useful too.

**Comment by davrichards** [comment]
OK thanks for the help and suggestions. Really appreciated.
